src/main/scala/org/scalajs/tools/tsimporter/sc/Definitions.scala

```diff
@@ -314,6 +314,18 @@ object TypeRef {
     }
   }
 
+  object Intersection {
+    def apply(types: List[TypeRef]): TypeRef =
+      TypeRef(QualifiedName.Root, types)
```

Comment:
`QualifiedName.Root` does not make any sense, here. There should be a dedicated `Name.INTERSECTION` with a pseudo-name, such as `"<with>"`, similar to `Name.SINGLETON`.

Reply:
Done in https://github.com/sjrd/scala-js-ts-importer/pull/46/commits/d5c796aa50da4cbcda53d82a363984bee8a9cc5a